src/control.f90

```diff
@@ -824,6 +826,23 @@ subroutine md_run (fixed_potential, vary_mu, total_energy)
 
     do iter = i_first, i_last ! Main MD loop
        mdl%step = iter
+       
+       ! At a given time step, if the difference between T_ext and the final temperature is larger than 
+       ! the temperature step (with 5% margin), then update T_ext 
+       ! Temperature evolves linearly from temp_ion to temp_ion_end over MDn_steps steps
+       ! Stop when target temperature has been reached
+       if  (abs(mdl%T_ext - temp_ion_end) > 0.95*abs((temp_ion_end-temp_ion)/dble(MDn_steps))) then
+         mdl%T_ext = temp_ion+(dble(iter-1)/dble(MDn_steps))*(temp_ion_end-temp_ion)
+         write(io_lun,fmt='(6x, "Thermostat temperature at step ", i5, ": ", f9.1, " K")') iter, mdl%T_ext 
+         ! Update target ke for SVR
+         thermo%ke_target = half*md_ndof_ions*fac_Kelvin2Hartree*mdl%T_ext
+         write(io_lun,fmt='(6x, "kee target is now" , f8.3)') thermo%ke_target
```

Comment:
This needs to be only on `ionode` (or `if(myid==0)` and only when `iprint` is high enough

Reply:
Should it be based on variable `iprint_gen` or `iprint_MD` ?

$\rightarrow$ `if(inode==ionode.AND.iprint_MD>2)` ?

Reply:
Since it's related to MD, I think that `iprint_MD` is better